skimage/filters/_gabor.py

```diff
@@ -72,35 +254,63 @@ def gabor_kernel(frequency, theta=0, bandwidth=1, sigma_x=None, sigma_y=None,
     >>> io.imshow(gk.real)  # doctest: +SKIP
     >>> io.show()           # doctest: +SKIP
     """
-    if sigma_x is None:
-        sigma_x = _sigma_prefactor(bandwidth) / frequency
-    if sigma_y is None:
-        sigma_y = _sigma_prefactor(bandwidth) / frequency
+    # handle deprecation
+    message = ('Using deprecated, 2D-only interface to gabor_kernel. '
+               'This interface will be removed in scikit-image 0.16. Use '
+               'gabor_kernel(frequency, sigma=(sigma_x, sigma_y)).')
+
+    if sigma_y is not None:
+        warn(message)
+        if 'sigma_x' in kwargs:
+            sigma = (sigma_y, kwargs['sigma_x'])
+        else:
+            sigma = (sigma_y, sigma)
+
+    # handle translation
+    if not isinstance(theta, coll.Iterable):
+        theta = (theta,) * (ndim - 1)
 
-    x0 = np.ceil(max(np.abs(n_stds * sigma_x * np.cos(theta)),
-                     np.abs(n_stds * sigma_y * np.sin(theta)), 1))
-    y0 = np.ceil(max(np.abs(n_stds * sigma_y * np.cos(theta)),
-                     np.abs(n_stds * sigma_x * np.sin(theta)), 1))
-    y, x = np.mgrid[-y0:y0 + 1, -x0:x0 + 1]
+    if not isinstance(sigma, coll.Iterable):
+        sigma = np.array([sigma] * ndim)
+    else:
+        sigma = np.append(sigma, [None] * (ndim - len(sigma)))
```

Comment:
Are you automagically padding sigma to match the number of dimensions? I would rather raise a ValueError when the number of elements in sigma doesn't match the dimensionality.

Reply:
@jni Yes. This was made as an extension of the functionality where you could provide only `sigma_x` or `sigma_y` and the other would be defaulted to `None` and later replaced with `default_sigma`.

However, I can just make a special deprecation case and jettison this padding anywhere else.